### DISCUSSION TITLE
transpile optionalChaining expressions while parsing

### DISCUSSION THREAD

**Discussion by Lookout-ent** [babel/babel#16214] [Q&A]
```
let babelParser = require('@babel/parser');
var ast = babelParser.parse(code);
```

input : var p = t?.[r].hostClass

is there any way that the parser would transpile optional chaining expressions and then produce ast ?

**Comment by nicolo-ribaudo** [comment]
No, the parser only produces the AST for the code as it is provided. You can traverse the AST generated by the parser to manually transpile optional chaining, or use `babel.transform` instead of `babelParser.parse` with the `ast: true` option (https://babeljs.io/docs/options#ast) so that it returns the transformed AST.

**Comment by Lookout-ent** [comment]
Thanks, this helped. I do have a follow-up though, right now we use babel-parser with a bunch of parser options, what are the equivalent options when using babel-core.transform()?

The parser options we currently use are as follows:
`let babelParserOption = {
        allowImportExportEverywhere: true,
        allowReturnOutsideFunction: true,
        allowAwaitOutsideFunction: true,
        allowUndeclaredExports: true,
        allowSuperOutsideMethod: true,
        attachComment: true,
        createParenthesizedExpressions: true,
        strictMode: false,
        sourceType: "unambiguous",
    }`

**Reply by nicolo-ribaudo** [reply]
```js
babel.transform(..., {
  parserOpts: babelParserOption
});
```

:)
